### DISCUSSION TITLE
Py code for Crocodile 10277

### DISCUSSION THREAD

**Discussion by PiotrSkut** [pybricks/support#1209] [Q&A]
Hey team,

Has anyone written code for the basic powered-up Hub to run the Lego crocodile train set 10277?

I want to power my crocodile train with this hub and motor
And control it with this remote, can i do that with PY bricks?

![image](https://github.com/pybricks/support/assets/142664325/b5199550-9145-4192-a435-6ef9749afb09)
![image](https://github.com/pybricks/support/assets/142664325/6a21780c-a14d-4257-9fe8-81e8bce2d8a3)
![image](https://github.com/pybricks/support/assets/142664325/aa72d4a3-e92d-4530-8594-43f368581927)

**Comment by laurensvalk** [comment]
Good to see you found the forum (Hi from Facebook!). Can you describe what each button should do exactly in your case?

Maybe +/- to increase speed by +/-10% and red button to stop? Anything else?

Here is a program that does that. There's a few comments that explain how you can tweak values like acceleration to your liking. Hope this works!

```python
from pybricks.pupdevices import Motor, Remote
from pybricks.parameters import Button, Port
from pybricks.tools import wait, StopWatch

train = Motor(Port.A)
remote = Remote()

# Get the settings for this motor.
max_speed, acceleration, torque = train.control.limits()

# Change 0.5 to something smaller for more realism.
train.control.limits(acceleration = acceleration * 0.5)

# Choose how many speeds steps you want.
NUM_STEPS = 10
STEP_NOW = 0

while True:

    # Wait for any button to be pressed.
    pressed = ()
    while not pressed:
        pressed = remote.buttons.pressed()
        wait(10)

    # Depending on what was pressed, get the speed level.
    if Button.RIGHT_PLUS in pressed:
        STEP_NOW += 1
    if Button.RIGHT_MINUS in pressed:
        STEP_NOW -= 1
    if Button.RIGHT in pressed:
        STEP_NOW = 0

    # Limit level to the maximum number of steps.
    STEP_NOW = max(-NUM_STEPS, min(STEP_NOW, NUM_STEPS))
    print("Speed level:", STEP_NOW)

    # Drive the train at the requested speed.
    if STEP_NOW:
        train.run(max_speed * STEP_NOW // NUM_STEPS)
    else:
        train.stop()

    # Wait for the button to be released.
    while pressed:
        pressed = remote.buttons.pressed()
        wait(10)
```

**Reply by PiotrSkut** [reply]
Morena Laurensvalk, yea at this stage the simple 10% speed increase or decrease with a press of the (+) or (-) buttons from 0 to 100%.  And an emergency stop on that red button in the middle. 

How would the pairing with the green button between the remote and hub work? any ideas?

**Reply by laurensvalk** [reply]
I've updated my answer with some example code. Let us know how it goes! Bonus points for a video that we can share on our site :smile:

**Reply by PiotrSkut** [reply]
Managed to get the code entered and it works, tha ks for the help.

**Reply by joeysant2010** [reply]
is there a code to run two motors off of the one hub?

**Reply by laurensvalk** [reply]
If you want to use regular motors (not train motors) as in the post above, you can just duplicate the lines with the motors:

Instead of

```python
train = Motor(Port.A)
```
you can do:

```python
train_front = Motor(Port.A)
train_rear = Motor(Port.B)
```

And then likewise in places where `train` was used, add another line. For example

```python
train.stop()
```

becomes

```python
train_front.stop()
train_rear.stop()
```

**Reply by BertLindeman** [reply]
> If you want to use regular motors (not train motors) as in the post above, you can just duplicate the lines with the motors:
> 
> Instead of
> 
> ```python
> train = Motor(Port.A)
> ```
> 
> you can do:
> 
> ```python
> train_front = Motor(Port.A)
> train_rear = Motor(Port.A)
> ```
> 
> And then likewise in places where `train` was used, add another line. For example
> 
> ```python
> train.stop()
> ```
> 
> becomes
> 
> ```python
> train_front.stop()
> train_rear.stop()
> ```

Laurens, I am confused. Twice a motor on the **same** port?

**Reply by laurensvalk** [reply]
Fixed the typo. Thanks!

**Comment by PiotrSkut** [comment]
Just a quick video with my phone now that uts free as I'm controlling the train with the remote!

**Reply by laurensvalk** [reply]
How it’s supposed to be 😄 

Glad you got it working. Video didn’t seem to get through though.

**Comment by PiotrSkut** [comment]
I've tried emailing it through... Can probably work out by now that IT is not my forte.

Seds
